rir/src/utils/configurations.cpp

```diff
@@ -23,54 +26,43 @@ void Configurations::parseINIFile() {
         //             "optimizations'\n";
         return defaultOptimizations();
     }
-    read(reader, "globalValueNumber");
-    read(reader, "escapeAnalysis");
-    read(reader, "forceDominance");
-    read(reader, "delayInstructions");
-    read(reader, "elideEnvironments");
-    read(reader, "delayEnvironments");
-    string cleanups = reader.Get("optimizations", "cleanup", "UNKNOWN");
-    if (cleanups != "UNKNOWN") {
-        std::stringstream data(cleanups);
-        std::string order;
-        while (std::getline(data, order, ',')) {
-            optimizations.insert(
-                new Optimization(new pir::Cleanup(), std::stoi(order)));
-        }
-    }
+
+    std::map<short, std::deque<pir::PirTranslator*>> opts;
+#define PARSE_OPT(Name, where, Klass)                                          \
+    if (short order = reader.GetInteger("optimizations", Name, 0))             \
+        opts[order].push_##where(new pir::Klass);
+
+    PARSE_OPT("forceDominance", front, ForceDominance());
+    PARSE_OPT("scopeResolution", front, ScopeResolution());
+    PARSE_OPT("delayInstructions", front, DelayInstr());
+    PARSE_OPT("elideEnvironments", front, ElideEnv());
+    PARSE_OPT("delayEnvironments", front, DelayEnv());
+    PARSE_OPT("inline", front, Inline());
+    PARSE_OPT("cleanup", back, Cleanup());
+    PARSE_OPT("cleanupSafepoints", back, CleanupSafepoint());
+#undef PARSE_OPT
+    for (auto& optlist : opts)
```

Comment:
Does this iterator respect the order?

Reply:
yes, opts is a `map`. maps are ordered by keys. the keys are of type short, so they will be ordered by increasing value. Then the `optlist.second` is a `deque`, which is sortof a `vector` you can insert from both sides. It is ordered too.